api/api.go

```diff
@@ -1000,6 +1000,10 @@ func (r *request) toHTTP() (*http.Request, error) {
 		return nil, err
 	}
 
+	if strings.HasPrefix(r.url.Host, "/") {
+		r.url.Host = "localhost"
```

Comment:
[Like the `moby` folks remarked, it's too bad we couldn't just leave this blank](https://github.com/moby/moby/pull/45942).

Reply:
So `.` or `-` may be accepted (see https://github.com/moby/moby/pull/45942#pullrequestreview-1529357346), but yeah, empty would've been most transparent.